github/repos.go

```diff
@@ -629,6 +629,50 @@ func (s *RepositoriesService) GetBranchProtection(ctx context.Context, owner, re
 	return p, resp, nil
 }
 
+// GetRequiredStatusChecks gets the required status checks for a given protected branch.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/branches/#get-required-status-checks-of-protected-branch
+func (s *RepositoriesService) GetRequiredStatusChecks(ctx context.Context, owner, repo, branch string) (*RequiredStatusChecks, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/branches/%v/protection/required_status_checks", owner, repo, branch)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches
+	req.Header.Set("Accept", mediaTypeProtectedBranchesPreview)
+
+	p := new(RequiredStatusChecks)
+	resp, err := s.client.Do(ctx, req, p)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return p, resp, nil
+}
+
+// ListRequiredStatusChecksContexts lists the required status checks contexts for a given protected branch.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/branches/#list-required-status-checks-contexts-of-protected-branch
+func (s *RepositoriesService) ListRequiredStatusChecksContexts(ctx context.Context, owner, repo, branch string) ([]string, *Response, error) {
```

Comment:
It might make the API more readable to use named return parameters here, since `[]string` is not as readable as a named type. What do you think of this?

```Go
func (s *RepositoriesService) ListRequiredStatusChecksContexts(ctx context.Context, owner, repo, branch string) (contexts []string, resp *Response, err error) {
```

You can look at [`MigrationService.MigrationArchiveURL`](https://godoc.org/github.com/google/go-github/github#MigrationService.MigrationArchiveURL) as some precedent, it returns `(url string, err error)` rather than `(string, error)`.

Reference: https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters.

However, it might be fine as is, since the method name `ListRequiredStatusChecksContexts` sorta implies that what's being returned are "contexts". But, the name is quite complex, so it might not be as obvious.

What do you think is better?